influxdb/server.sls

```diff
@@ -25,8 +25,70 @@ influxdb_service:
   service.running:
   - enable: true
   - name: {{ server.service }}
+  # This delay is needed before being able to send data to server to create
+  # users and databases.
+  - init_delay: 5
   - watch:
     - file: influxdb_config
     - file: influxdb_default
 
+{% set url_for_query = "http://{}:{}/query".format(server.http.bind.address, server.http.bind.port) %}
+
+{%- if server.admin.user is defined %}
+{% set query_create_admin = "--data-urlencode \"q=CREATE USER {} WITH PASSWORD '{}' WITH ALL PRIVILEGES\"".format(server.admin.user.name, server.admin.user.password) %}
+influxdb_create_admin_{{server.admin.user.name}}:
+  cmd.run:
+  - name: curl -s -POST "{{ url_for_query }}" {{ query_create_admin }}
```

Comment:
You should use `-f` flag to fail if server returns error, maybe also `-S` to print error if this happens.